zerver/tests/test_import_export.py

```diff
@@ -551,6 +551,23 @@ def get_custom_profile_field_names(r: Realm) -> Set[str]:
 
         assert_realm_values(get_custom_profile_field_names)
 
+        def get_custom_profile_with_field_type_user(r: Realm) -> Set[str]:
+            custom_profile_fields = CustomProfileField.objects.filter(
+                field_type=CustomProfileField.USER,
+                realm=r)
+            custom_profile_field_values = []  # type: List[CustomProfileFieldValue]
+            for field in custom_profile_fields:
+                custom_profile_field_values += CustomProfileFieldValue.objects.filter(field=field)
+
+            user_emails = set()
+            for field in custom_profile_field_values:
+                user_id_list = ujson.loads(field.value)
+                for user_id in user_id_list:
+                    user_emails.add(UserProfile.objects.get(id=user_id).email)
+            return user_emails
```

Comment:
I think in about the same amount of code you can verify a more structured result than just a big set of emails that span all field values (and which probably have overlap).  Can you look at this code and see if it makes sense?

~~~ py
fields = CustomProfileFields.objects.filter(
    field_type=CustomProfileField.USER,
    realm=r)

def get_email(user_id):
    return UserProfile.objects.get(id=user_id).email

def get_emails_from_value(value):
    user_id_list = ujson.load(value.value)
    return {
        get_email(user_id)
        for user_id in user_id_list}

def values_for(field):
    values = CustomProfileFieldValue.objects.filter(field)
    return {
        (get_email(value.user_profile_id),
        get_emails_from_value(value)
        for value in values}

return {
    (field.name, field.hint, values_for(field))
    for field in fields}

Reply:
Actually we are not allowed to have set of sets in python. Although I like this structure, so I'll modify the code to return Tuple(set of emails, set of field names, set of field hints)